Automattic/wp-calypso
# Marketplace: Add MailPoet confirmation page

Solves https://github.com/Automattic/dotcom-forge/issues/3305

## Proposed Changes

* Add a new page under `my-sites/plugins` that displays a confirmation message for MailPoet subscription
* When the `Get it now` button is clicked, the `/sites/${ siteId }/mailpoet/add-upgrade` endpoint is called
* When the request completes, a notification of success or failure is displayed, and the buttons change to only one button, `Go back` that goes to previous page.
* If the user clicks on the `No, I'm ok` button, the system goes to previous page.

|Before | Error| Success|
|-------|------|------|
|      ![CleanShot 2023-09-26 at 16 12 03@2x](https://github.com/Automattic/wp-calypso/assets/3519124/27142623-c4d5-4944-8334-3fa2cecc3062)      |       ![CleanShot 2023-09-26 at 16 12 12@2x](https://github.com/Automattic/wp-calypso/assets/3519124/1350dc0e-8e78-4471-9a0b-0132822f08e9)      |     ![CleanShot 2023-09-26 at 16 14 15@2x](https://github.com/Automattic/wp-calypso/assets/3519124/49ad9f76-a557-4ac6-80d5-a91aa54716c1)       |

## Testing Instructions



* There is not yet an entry-point for this page, but it can be accessed by URL
* Navigate to `/plugins/mailpoet-business/upgrade/:site_id`, selecting a `:siteId` with the `eCommerce` plan
* Click on `Get it now`
* Make sure that you can see a success notification with the message `The MailPoet is added to your site. MailPoet plugin will be installed and activated shortly.`
* Make sure that when the notification is displayed, the buttons change to only one button reading `Go back`

- Navigate to `/plugins/mailpoet-business/upgrade/:site_id`, selecting a `:siteId` with the same site in the previous step
- Click on `Get it now`
- Make sure you can see an error notification with the message: `This site already has a MailPoet upgrade.`

* Navigate to `/plugins/mailpoet-business/upgrade/:site_id`, selecting a `:siteId` with a plan lower than `eCommerce
* Click on `Get it now`
* Make sure you can see an error notification with a message indicating that the site does not have the required plan

- Navigate to the plans page `/plans/:siteId` with any `siteId`
- Navigate to `/plugins/mailpoet/confirmation/:site_id` with any `siteId`
- Click on the `Go back` button
- Make sure the system navigates back to the plans page

## Pre-merge Checklist
